script.js

```diff
@@ -24,8 +24,14 @@ isChrome = () => {
 };
 
 headersReceivedListener = (requestDetails) => {
+  // Do not count bytes from requests from local cache
+  if (requestDetails.fromCache) return
+
+  let CONST_INCOGNITO = 'Incognito';
+  
   if (isChrome()) {
-     const origin = extractHostname(!requestDetails.initiator ? requestDetails.url : requestDetails.initiator);
+    // If Incognito request we do not track the domain origin
+     const origin = (requestDetails.incognito ? CONST_INCOGNITO : extractHostname(!requestDetails.initiator ? requestDetails.url : requestDetails.initiator));
```

Comment:
It's easier to read like this I think (without negation)
```suggestion
     const origin = (requestDetails.incognito ? CONST_INCOGNITO : extractHostname(requestDetails.initiator ? requestDetails.initiator : requestDetails.url));
```